### PR TITLE
[SPARK-29145][SQL][FOLLOWUP] Clean up code about support sub-queries in join conditions

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
According to discuss https://github.com/apache/spark/pull/25854#discussion_r629451135

### Why are the changes needed?
Clean code

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existed UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
FYI @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 1]
```diff
@@ -2353,7 +2353,7 @@ class Analyzer(override val catalogManager: CatalogManager)
       case q: UnaryNode if q.childrenResolved =>
         resolveSubQueries(q, q.children)
       case j: Join if j.childrenResolved =>
-        resolveSubQueries(j, Seq(j, j.left, j.right))
+        resolveSubQueries(j, Seq(j.left, j.right))
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:2356]: nit: `j.children`

Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala]: > nit: `j.children`

Done

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 2]
```diff
@@ -2353,7 +2353,7 @@ class Analyzer(override val catalogManager: CatalogManager)
       case q: UnaryNode if q.childrenResolved =>
         resolveSubQueries(q, q.children)
       case j: Join if j.childrenResolved =>
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:2355]: seems we should also check `j.duplicateResolved`

Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala]: Done, but I am confused witch case will cause error.

**Comment by maropu** [comment]
Is the title correct? It seems a clean-up fix.

**Comment by AngersZhuuuu** [comment]
> Is the title correct? It seems a clean-up fix.

Hmmm how about current?

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**